action.yml

```diff
@@ -1,23 +1,72 @@
-name: 'Example composite GitHub action'
-description: 'Example composite GitHub action'
+name: 'GitHub Action Atmos Component Updater'
+description: 'GitHub Action Atmos Component Updater'
 author: hello@cloudposse.com
 branding:
-  icon: 'file'
+  icon: 'server'
   color: 'white'
 inputs:
-  param1:
+  token:
     required: true
-    description: "Input parameter placeholder"
-    default: "true"
-outputs:
-  result1:
-    description: "Output result placeholder"
-    value: "${{ steps.context.outputs.action-result }}"
+    description: "GitHub Private Access Token"
+  desired-components-version:
+    required: false
+    description: "Desired Components Version. If not set it gets latest tag from cloudposse/terraform-aws-components repo"
+    default: ''
+  infra-repo-name:
+    required: true
+    description: "Infrastructure repository name"
+  infra-repo-dir:
+    required: true
+    description: "Infrastructure repository directory"
 runs:
   using: "composite"
   steps:
-    - id: context
+    - name: "Setup Terraform"
+      uses: hashicorp/setup-terraform@v2
+
+    - name: "Setup Atmos"
+      uses: cloudposse/github-action-setup-atmos@v1.0.0
+
+    - name: "Setup Python"
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.10'
+
+    - name: "Install Python Dependencies"
+      shell: bash
+      run: |
+        cd ${{ github.action_path }}
+        pip install -r src/requirements.txt
+
+    - if: ${{ inputs.desired-components-version == '' }}
+      name: "Checkout 'terraform-aws-components' Repository"
+      uses: actions/checkout@v3
+      with:
+        repository: cloudposse/terraform-aws-components
```

Comment:
Parameterize

Reply:
This actually won't work, since the source of the components is in the `component.yaml`

Reply:
I'm not sure what we should do here. If we should move this to a separate task or try to solve it.

Reply:
ok, looks like i misunderstood requirements. I thought that this action should support components only from `cloudposse/terraform-aws-components` repo.

Reply:
will rework